src/AutoDiff/tests/CppADTest.cpp

```diff
@@ -0,0 +1,44 @@
+/**
+ * @file CppADTest.cpp
+ * @authors Giulio Romualdi
+ * @copyright 2020 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+// Catch2
+#include <catch2/catch.hpp>
+
+#include <BipedalLocomotion/AutoDiff/CppAD.h>
+
+TEST_CASE("CppAD and Eigen")
+{
+    // Some coefficients useful for the test
+    constexpr double a = 4;
+    constexpr double b = 1.1341;
+    constexpr double c = 2.3213;
+
+    BipedalLocomotion::AutoDiff::VectorXcppAD x(3);
+
+    // Start recording
+    CppAD::Independent(x);
+    BipedalLocomotion::AutoDiff::VectorXcppAD y = a * x.array() + c * (b * x.array()).sin();
```

Comment:
To be honest ``VectorXcppAD`` seems a little ugly to me (like a fiscal code :grin:). Does it make sense to use an additional namespace, such that we have
```
BipedalLocomotion::AutoDiff::cppAD::VectorXAD
```
? Does it look any better in your opinion?

Reply:
Done in 2814e3bb6322b95f1e05afaec30c16d6d98ad771